packages/shared/src/utils/os/os_detect.ts

```diff
@@ -1,16 +1,27 @@
-declare global {
-    interface Window {
-        opera?: string;
-        MSStream?: {
-            readonly type: string;
-            msClose: () => void;
-            msDetachStream: () => void;
-        };
+/**
+ * Requests detailed data with User-Agent Client Hints
+ *
+ * @see https://wicg.github.io/ua-client-hints/
+ * @returns {Promise<UADataValues | null | undefined>} Promise with detailed data or null or undefined if not supported
+ * **/
+const getUserAgentData = async () => {
+    if ('userAgentData' in navigator) {
+        const user_agent_data = await navigator.userAgentData?.getHighEntropyValues([
```

Comment:
I saw that this is an experimental feature which doesn't support some browsers like Firefox and Safari ( according to the [doc](https://developer.mozilla.org/en-US/docs/Web/API/NavigatorUAData/getHighEntropyValues#browser_compatibility) )

So for such devices we are returning `unknown OS` inside `OSDetect()` ? If so, is this a expected behavior that we want?

Reply:
No @shayan-deriv, we don't return `unknown OS` unless the previous code was unable to determine the `OS`. As seen in this line of code `const user_platform = user_agent_data ? user_agent_data.platform : navigator?.platform ?? '';` inside of  `OSDetect`, we check if the `returned` value of `getUserAgentData` (`user_agent_data`) is `truthy`, then we use the `platform` property's value as the value, otherwise, we fallback to the previous code.

Reply:
got did :+1:  thanks for the explanation :pray: